### DISCUSSION TITLE
Track gas for read-only methods

### DISCUSSION THREAD

**Discussion by razgraf** [sablier-labs/evm-monorepo#37] [Ideas]
Related to PR #31 we should start tracking (probably using data from foundry) gas consumption for our heavy read-only methods.

In case we decide to have a light version of the app that only relies on web3 calls, we'll make use of multicalls to batch queries to the RPC. These RPC nodes will probably have limits for how much gas a read-only method can consume. Therefore, by figuring our the gas limit and the gas required to gather data for one stream, we can optimize the number of sequential queries needed in the web3 native version of the app.

P.S. Worth looking into this version of [Multicall 3](https://github.com/mds1/multicall) which might be more efficient on reads.

**Comment by PaulRBerg** [comment]
We could use Forge's [gas snapshots](https://book.getfoundry.sh/forge/gas-snapshots) for this.

**Comment by PaulRBerg** [comment]
Aaand we are now indeed using gas snapshot to instrument the gas usage of the test functions.

Locking.
